src/pages/countries.js

```diff
@@ -0,0 +1,82 @@
+import { useQuery, gql } from "@apollo/client";
+import { Link, useLocation, useHistory } from "react-router-dom";
+import CountriesList from "../components/CountriesList";
+import { useState, useEffect } from "react";
+
+export const COUNTRIES_QUERY = gql`
+  query allCountries($limit: Int, $skip: Int) {
+    countries(limit: $limit, skip: $skip) {
+      id
+      name
+      vatRate
+      languages {
+        id
+        name
+      }
+    }
+  }
+`;
+
+function useQueryUrl() {
+  return new URLSearchParams(useLocation().search);
+}
+
+const Countries = (props) => {
+  const query = useQueryUrl();
+  const history = useHistory();
+
+  const countriesPerPage = 5;
+  const [page, setPage] = useState(query.get("page"));
+
+  useEffect(() => {
+    setPage(query.get("page"));
+  }, [query.get("page")]);
+
+  const variables = {
+    variables: { limit: countriesPerPage, skip: page * countriesPerPage },
+  };
+
+  const { loading, error, data } = useQuery(COUNTRIES_QUERY, variables);
+
+  const getNextPageNumber = () => {
+    const currPage = parseInt(query.get("page")) || 0;
+    const nextPage = currPage + 1;
+    return nextPage;
+  };
+  const getPreviousPageNumber = () => {
+    const currPage = parseInt(query.get("page"));
+    if (currPage >= 1) {
+      const previousPage = currPage - 1;
+      return previousPage;
+    } else {
+      return 0;
```

Comment:
I think that this validation logic should be in handleClick function because in the current situation we can do `history.push` a lot of time with `0` page, better to prevent history pushing at all in that case

Reply:
I remodeled functions  for handling next and previous page. Now I only have `handleClick` functions without  `getPreviousPageNumber` etc.